nemo_skills/pipeline/eval.py

```diff
@@ -190,34 +226,43 @@ def eval(
 
     eval_cmds = (
         [
-            get_greedy_cmd(
+            cmd
+            for benchmark in benchmarks.keys()
+            for cmd in get_greedy_cmd(
                 benchmark,
                 split,
                 output_dir,
                 extra_eval_args=extra_eval_args,
                 extra_arguments=extra_arguments,
                 extra_datasets=extra_datasets,
+                num_chunks=num_chunks,
+                chunk_ids=chunk_ids,
             )
-            for benchmark in benchmarks.keys()
         ]
         if not skip_greedy
         else []
     )
     eval_cmds += [
-        get_sampling_cmd(
+        cmd
+        for benchmark, rs_num in benchmarks.items()
+        for rs in range(starting_seed, starting_seed + rs_num)
+        for cmd in get_sampling_cmd(
             benchmark,
             split,
             output_dir,
             rs,
             extra_eval_args=extra_eval_args,
             extra_arguments=extra_arguments,
             extra_datasets=extra_datasets,
+            num_chunks=num_chunks,
+            chunk_ids=chunk_ids,
         )
-        for benchmark, rs_num in benchmarks.items()
-        for rs in range(starting_seed, starting_seed + rs_num)
     ]
     if num_jobs == -1:
         num_jobs = len(eval_cmds)
+    else:
```

Comment:
Yes its generally the final number of jobs being launched

Reply:
The thing here is we have num_jobs which is distributing benchmarks / samples on different nodes and now also chunks, which split a single benchmark across different nodes (and we restrict the same chunks for each benchmark, otherwise it's too many options). So currently we will have num_jobs * num_chunks total jobs but it's a bit confusing, so maybe some renaming or some other changes are needed